ExampleMod/Projectiles/ExampleLastPrismBeam.cs

```diff
@@ -0,0 +1,353 @@
+			// Add a fixed offset to align with the Prism's sprite sheet.
+			projectile.position += hostPrismDir * 16f + new Vector2(0f, -hostPrism.gfxOffY);
+			// Add the forwards offset, measured in pixels.
+			projectile.position += hostPrismDir * beamStartForwardsOffset;
+			// Add the sideways offset vector, which is calculated for the current angle of the beam and scales with the beam's sideways offset.
+			projectile.position += beamSpanVector;
+
+			// Set the beam's velocity to point towards its current spread direction and sanity check it. It should have magnitude 1.
+			projectile.velocity = hostPrismDir.RotatedBy(sinusoidYOffset);
+			if (projectile.velocity.HasNaNs() || projectile.velocity == Vector2.Zero)
+				projectile.velocity = -Vector2.UnitY;
+			projectile.rotation = projectile.velocity.ToRotation();
+
+			// Update the beam's length by performing a hitscan collision check.
+			float hitscanBeamLength = PerformBeamHitscan(hostPrism, chargeRatio >= 1f);
+			BeamLength = MathHelper.Lerp(BeamLength, hitscanBeamLength, BeamLengthChangeFactor);
+
+			// This Vector2 stores the beam's hitbox statistics. X = beam length. Y = beam width.
+			Vector2 beamDims = new Vector2(projectile.velocity.Length() * BeamLength, projectile.width * projectile.scale);
+
+			// Only produce dust and cause water ripples if the beam is above a certain charge level.
+			Color beamColor = GetBeamColor();
+			if (chargeRatio >= VisualEffectThreshold)
+			{
+				ProduceBeamDust(beamColor);
+
+				// If the game is rendering (i.e. isn't a dedicated server), make the beam disturb water.
+				if (Main.netMode != NetmodeID.Server)
+					ProduceWaterRipples(beamDims);
+			}
+
+			// Make the beam cast light along its length. The brightness of the light scales with the charge.
+			// v3_1 is an unnamed decompiled variable which is the color of the light cast by DelegateMethods.CastLight
+			DelegateMethods.v3_1 = beamColor.ToVector3() * BeamLightBrightness * chargeRatio;
+			Utils.PlotTileLine(projectile.Center, projectile.Center + projectile.velocity * BeamLength, beamDims.Y, new Utils.PerLinePoint(DelegateMethods.CastLight));
+		}
+
+		// Uses a simple polynomial (x^3) to get sudden but smooth damage increase near the end of the charge-up period.
+		private float GetDamageMultiplier(float chargeRatio)
+		{
+			float f = chargeRatio * chargeRatio * chargeRatio;
+			return MathHelper.Lerp(1f, MaxDamageMultiplier, f);
+		}
+
+		private float PerformBeamHitscan(Projectile prism, bool fullCharge)
+		{
+			// By default, the hitscan interpolation starts at the projectile's center.
+			// If the host Prism is fully charged, the interpolation starts at the Prism's center instead.
+			// Overriding that, if the player shoves the Prism into or through a wall, the interpolation starts at the player's center.
+			// This last part prevents the player from projecting beams through walls under any circumstances.
+			Vector2 samplingPoint = projectile.Center;
+			if (fullCharge)
```

Comment:
This is the first occurence of doing "one line" if statements on the next line. There are countless ones in your code. It's not something examplemod does, afaik it just does 
```cs
if (fullCharge) {
    samplingPoint = prism.Center;
}

Reply:
I use them in tML, and there'll be a lot in the Terraria source soon, but I tend to agree, keeping ExampleMod all braces helps newbies avoid pitfalls. Braceless is contentious as is

Reply:
Agreed. While it is necessary for people to learn to read braceless code (so I'm okay with TML internals and the TML formatting of Terraria source having it for example), ExampleMod is meant to be a starting point and shouldn't have potentially confusing choices in it.